modules/sdk-core/src/bitgo/utils/opengpgUtils.ts

```diff
@@ -156,34 +157,53 @@ export async function createSharedDataProof(
 }
 
 /**
- * Creates an Eddsa KeyShare Proof by appending an ed25519 subkey to an armored gpg private key.
+ * Creates a KeyShare Proof based on given algo.
+ *
+ * Creates an EdDSA KeyShare Proof by appending an ed25519 subkey to an armored gpg private key.
+ * Creates an EcDSA KeyShare Proof by Append a secp256k1 subkey (auth) to a PGP keychain.
+ *
+ * For EdDSA:
+ * uValue itself is used to construct by EDDSA signature
+ * the proof is however generated via ECDSA sinature
+ * proof != Value
+ *
+ * For EcDSA:
+ * both proof and signature are generated by EcDSA algorithm
  *
  * @param privateArmor gpg private key in armor format
  * @param uValue u value from an Eddsa keyshare
+ * @param algo algo to use, eddsa or ecdsa
  * @return {string} keyshare proof
  */
-export async function createShareProof(privateArmor: string, uValue: string): Promise<string> {
+export async function createShareProof(privateArmor: string, uValue: string, algo: string): Promise<string> {
   const privateKey = await readKey({ armoredKey: privateArmor });
   const dateTime = new Date();
-  await sodium.ready;
-  const subKeyVal = Buffer.from(
-    sodium.crypto_scalarmult_ed25519_base_noclamp(Buffer.from(uValue, 'hex'), 'uint8array')
-  );
-
-  // Sub-key (encryption key) packet.
-  const publicSubKey = new Uint8Array([...new Uint8Array([0x40]), ...new Uint8Array(subKeyVal)]);
-  const subOid = [0x2b, 0x06, 0x01, 0x04, 0x01, 0xda, 0x47, 0x0f, 0x01];
-
-  // @ts-ignore
-  subOid.write = () => new Uint8Array(Buffer.from('092b06010401da470f01', 'hex'));
   // @ts-ignore - type inconsistency, this ctor supports a date param: https://docs.openpgpjs.org/SecretSubkeyPacket.html
   const secretSubkeyPacket = new pgp.SecretSubkeyPacket(dateTime);
-  secretSubkeyPacket.algorithm = pgp.enums.publicKey.eddsa;
+  secretSubkeyPacket.algorithm = pgp.enums.publicKey[algo];
   // @ts-ignore - same as above
   secretSubkeyPacket.isEncrypted = false;
+  let oid;
+  let Q;
+  if (algo === 'eddsa') {
+    await sodium.ready;
+    const subKeyVal = Buffer.from(
+      sodium.crypto_scalarmult_ed25519_base_noclamp(Buffer.from(uValue, 'hex'), 'uint8array')
+    );
+    // Sub-key (encryption key) packet.
+    oid = [0x2b, 0x06, 0x01, 0x04, 0x01, 0xda, 0x47, 0x0f, 0x01];
+    // @ts-ignore
+    oid.write = () => new Uint8Array(Buffer.from('092b06010401da470f01', 'hex'));
+    Q = new Uint8Array([...new Uint8Array([0x40]), ...new Uint8Array(subKeyVal)]);
```

Comment:
you dont need the first `...`
it could be instead: `[new Uint8Array([0x40])`

Reply:
cannot remove it, overloading prevents me from this

Reply:
why would spread syntax be required?   you can absolutely do this (i do this already in hsm-api):
`Q = new Uint8Array([new Uint8Array([0x40]), ...new Uint8Array(subKeyVal)]);`

Reply:
you have in JS, but i have in TS. 
![image](https://user-images.githubusercontent.com/95242722/213250038-9f9d75bf-ec5c-436c-81e5-f566c8e71d01.png)

Reply:
why do this anyways `...new Uint8Array([0x40]) ` then, when we can do ` [0x40, ...new Uint8Array(subKeyVal)]`

Reply:
@Logicwax is there a reason why you do an array of an array and numbers, that is [[0x40], x, y, z] in instead of [0x40, x, y, z]? Is the former required by openpgp?

Reply:
```
    const subKeyVal = Buffer.from(sodium.crypto_scalarmult_ed25519_base_noclamp(Buffer.from(uValue, 'hex')));
    Q = new Uint8Array([0x40, ...subKeyVal]);
```
or the simplest, i think they comes down to the same flat array. 
so they really doesnt make much difference?

Reply:
@islamaminBitGo yeah you're right, that works too.

@jicli this could simply be `Q = new Uint8Array([0x40, ...subKeyVal])`   does that work in TS?

Reply:
yes it would work, updated